### DISCUSSION TITLE
Have you considered using Embeddings with Elasticsearch for more relevant results?

### DISCUSSION THREAD

**Discussion by NOT-HAL9000** [magda-io/magda#3400] [Ideas]
Hey Everyone,

I was having a look through data.gov.au and I found the basic search results to be pretty unusable. 
Would it be a bad idea to explore using document embeddings + vector similarity in elastic in order to surface even more relevant results?
See elastic article on it here: [Text Embeddings and Vector Search with elastic](https://www.elastic.co/blog/how-to-deploy-nlp-text-embeddings-and-vector-search)

**Comment by t83714** [comment]
To make the search engine "smarter", we are actually looking at generating a knowledge graph (neo4j based) from datasets using existing NLP models and make recommendations for datasets of "relevant concept".
We created some prototypes but might take some time to make it useful. 
Thanks for your suggestions --- I think vector search could be an option for us to introduce some short term improvement.
